jupyter/d2l-java/chapter_multilayer-perceptrons/dropout.ipynb

```diff
@@ -0,0 +1,785 @@
+    "according to the specified dropout probability.\n",
+    "When the model is not in training mode,\n",
+    "the `Dropout` layer simply passes the data through during testing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "6"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "SequentialBlock net = new SequentialBlock();\n",
+    "net.add(Blocks.batchFlattenBlock(784));\n",
+    "net.add(Linear.builder().setOutChannels(256).build());\n",
+    "net.add(Activation::relu);\n",
+    "net.add(Dropout.builder().optProbability(dropout1).build());\n",
+    "net.add(Linear.builder().setOutChannels(256).build());\n",
+    "net.add(Activation::relu);\n",
+    "net.add(Dropout.builder().optProbability(dropout2).build());\n",
+    "net.add(Linear.builder().setOutChannels(10).build());\n",
+    "net.setInitializer(new NormalInitializer(0.01f));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we train and test the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "7"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "float epochLoss = 0f;\n",
+    "float accuracyVal = 0f;\n",
+    "\n",
+    "Model model = Model.newInstance(\"MLP\");\n",
+    "model.setBlock(net);\n",
+    "\n",
+    "LearningRateTracker lrt = LearningRateTracker.fixedLearningRate(0.5f);\n",
+    "Optimizer sgd = Optimizer.sgd().setLearningRateTracker(lrt).build();\n",
+    "\n",
+    "Loss loss = Loss.softmaxCrossEntropyLoss();\n",
+    "\n",
+    "DefaultTrainingConfig config = new DefaultTrainingConfig(loss)\n",
+    "                .optOptimizer(sgd) // Optimizer (loss function)\n",
+    "                .addEvaluator(new Accuracy()) // Model Accuracy\n",
+    "                .addTrainingListeners(TrainingListener.Defaults.logging()); // Logging\n",
+    "\n",
+    "Trainer trainer = model.newTrainer(config);\n",
+    "trainer.initialize(new Shape(1, 784));\n",
+    "\n",
+    "for (int epoch = 1; epoch <= numEpochs; epoch++) {\n",
+    "        // Iterate over dataset\n",
+    "        for (Batch batch : trainer.iterateDataset(trainIter)) {\n",
+    "            // Update loss and evaulator\n",
+    "            EasyTrain.trainBatch(trainer, batch);\n",
+    "            \n",
+    "            // Update parameters\n",
+    "            trainer.step();\n",
+    "\n",
+    "            batch.close();\n",
+    "        }\n",
+    "\n",
+    "        for (Batch batch : trainer.iterateDataset(testIter)) {\n",
+    "            // Update loss and evaulator\n",
+    "            EasyTrain.validateBatch(trainer, batch);\n",
+    "\n",
+    "            batch.close();\n",
+    "        }\n",
+    "    \n",
+    "        trainer.notifyListeners(listener -> listener.onEpoch(trainer));\n",
```

Comment:
We should try to use EasyTrain.fit here and in a few other files in this PR. In general, we should use EasyTrain.fit wherever the book doesn't include the full verbose training loops itself.

If you look at the metrics (trainer.getMetrics()), you should be able to find the recorded evaluation data you are trying to plot recorded by the EvaluatorTrainingListener. You can find the metric name with EvaluatorTrainingListener.metricName. You may also need to call trainer.setMetrics(new Metrics()) to enable metric recording